confluent_kafka/consumer.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from confluent_kafka.cimpl import (KafkaError,
+                                   Consumer as _cConsumer)
+from .error import ConsumeException
+from .serialization import (SerializationError,
+                            SerializationContext,
+                            MessageField)
+
+
+class DeserializingConsumer(_cConsumer):
+    """
+    A client that consumes records from a Kafka cluster. With Serialization
+    capabilities.
+
+    Args:
+        conf (ClientConfig): Client configuration
+            The following configurations are supported in addition to the ones
+            described in Client Configurations(linked below).
+
+            key.deserializer (Deserializer): Deserializes
+                :py:func:`Message.key()` return value.
+
+            value.deserializer (Deserializer): Deserializes
+                :py:func:`Message.value()` return value.
+
+            error_cb(callable(KafkaError), optional): Callback for
+                generic/global error events. These errors are typically to be
+                considered informational since the client will automatically try
+                to recover. This callback is served upon calling
+                :py:func:`Consumer.poll()`
+
+            log_cb (logging.Handler, optional): logging handle to forward logs
+                to. To avoid spontaneous calls from non-Python threads the log
+                messages will only be forwarded when :py:func:`Consumer.poll()`
+                is called.
+
+            stats_cb (callable(str), optional): Callback for statistics data.
+                This callback is triggered by :py:func:`Consumer.poll()` every
+                ``statistics.interval.ms`` (needs to be configured separately).
+                The str function argument is a str instance of a JSON document
```

Comment:
i don't like the word 'document' here. maybe "Statistics are provided as a JSON formatted string"? i'd also remove the last sentence as you already mention this, and i don't like the word serve (i'm not sure it's meaning is well understood in the target community).